src/uucore/src/lib/features/splice.rs

```diff
@@ -0,0 +1,84 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+//! This module provides the [`write_fast_using_splice`] function to leverage the `splice` system call
+//! in Linux systems, thus increasing the I/O performance of copying between two file descriptors.
+
+use nix::unistd;
+use std::{
+    io::Read,
+    os::{
+        fd::AsFd,
+        unix::io::{AsRawFd, RawFd},
+    },
+};
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+use crate::pipes::{pipe, splice, splice_exact};
+
+const SPLICE_SIZE: usize = 1024 * 128;
+const BUF_SIZE: usize = 1024 * 16;
+
+/// `splice` is a Linux-specific system call used to move data between two file descriptors without
+/// copying between kernel and user spaces. This results in a large speedup.
+///
+/// This function reads from a file/stream `handle` and directly writes to `write_fd`. Returns the
+/// amount of bytes written as a `u64`.
+///
+/// The `bool` in the result value indicates if we need to fall back to normal
+/// copying or not. False means we don't have to.
+#[inline]
+pub fn write_fast_using_splice<R: Read + AsFd + AsRawFd, S: AsRawFd + AsFd>(
+    handle: &R,
+    write_fd: &S,
+) -> nix::Result<(usize, bool)> {
```

Comment:
would it be possible to use UError https://docs.rs/uucore/0.0.28/uucore/error/trait.UError.html ?